taf/updater/updater.py

```diff
@@ -279,7 +280,18 @@ def update_repository(config: RepositoryConfig):
         if config.url is None:
             raise UpdateFailedError("URL cannot be determined. Please specify it")
 
-    return _update_or_clone_repository(config)
```

Comment:
I don't think that should be necessary this line. `_update_or_clone_repository` performs `validation`, so no need to call both. We also need to keep the `return` here. We collect and return some data that is then used by out lifecycle handlers. They allow the users to get access to the information about the update that was run (was it successful, what was pulled etc.)